collect_app/src/main/java/org/odk/collect/android/activities/viewmodels/FormDownloadListViewModel.java

```diff
@@ -163,7 +164,7 @@ public void setUrl(String url) {
     }
 
     public String[] getFormIdsToDownload() {
-        return formIdsToDownload;
+        return Arrays.copyOf(formIdsToDownload, formIdsToDownload.length);
```

Comment:
The code related to external apps is relatively new and a bit fishy to me. I don't want to touch it here because it would require testing again. There is still a pr on the collectTester's side https://github.com/grzesiek2010/collectTester/pulls I'll take a look at it then.